src/main/java/com/segment/analytics/android/integrations/optimizelyx/OptimizelyXIntegration.java

```diff
@@ -1,63 +1,102 @@
+import com.optimizely.ab.config.Variation;
+
+import java.util.Map;
 
 /**
- * Kahuna helps mobile marketers send push notifications and in-app messages.
+ * Optimizely X helps marketers and growth hackers make smart decisions through A/B testing.
  *
- * @see <a href="https://www.optimizely.com/">Pptimizely</a>
- * @see <a href="https://segment.com/docs/integrations/optimizely/">Optimizely Integration</a>
- * @see <a href="http://developers.optimizely.com/android/">Optimizely Android SDK</a>
+ * @see <a href="https://www.optimizely.com/">Optimizely</a>
+ * @see <a href="https://segment.com/docs/destinations/optimizelyx/">Optimizely Integration</a>
+ * @see <a
+ *     href="https://developers.optimizely.com/x/solutions/sdks/introduction/index.html?language=android&platform=mobile">Optimizely
+ *     Android SDK</a>
  */
-public class OptimizelyXIntegration extends Integration<Void> {
-  private static final String OPTIMIZELY_KEY = "Optimizely";
-
-  private final Context context;
-  private final Logger logger;
-  private final Analytics analytics;
-
-  // Optimizely must be initialized immediately on launch. So we require the token when creating the
-  // factory, and initialize it in this method.
-//  public static Factory createFactory(final Application application, String token) {
-////    Optimizely.startOptimizelyWithAPIToken(token, application);
-////    return new Factory() {
-////      @Override public Integration<?> create(ValueMap settings, Analytics analytics) {
-////        boolean listen = settings.getBoolean("listen", false);
-////        Logger logger = analytics.logger(OPTIMIZELY_KEY);
-////
-////        OptimizelyIntegration integration =
-////            new OptimizelyIntegration(analytics, application, logger);
-////        if (listen) {
-////          Optimizely.addOptimizelyEventListener(integration);
-////        }
-////        return integration;
-////      }
-//
-////      @Override public String key() {
-////      return OPTIMIZELY_KEY;
-//////      }
-//    };
-//  }
-
-  public OptimizelyXIntegration(Analytics analytics, Context context, Logger logger) {
-    this.analytics = analytics;
-    this.context = context;
-    this.logger = logger;
-  }
+public class OptimizelyXIntegration extends Integration<NotificationListener> {
+  static final Options options = new Options().setIntegration("Optimizely X", false);
+  private static final String OPTIMIZELYX_KEY = "Optimizely X";
+  final NotificationListener listener;
+
+  public static final Factory FACTORY =
+      new Factory() {
+        @Override
+        public Integration<?> create(ValueMap settings, Analytics analytics) {
+          return new OptimizelyXIntegration(analytics);
+        }
+
+        @Override
+        public String key() {
+          return OPTIMIZELYX_KEY;
+        }
+      };
+
+  OptimizelyXIntegration(final Analytics analytics) {
+
+    listener =
+        new NotificationListener() {
+          @Override
+          public void onEventTracked(
```

Comment:
If I understood this correctly, this listener on iOS listens for an Optimizely event to be tracked - which we are not concerned about. I think we need to only listen for the onExperimentActivated event since the main use case that customers are looking for is the ability to send events to Optimizely without having to instrument  native Optimizely track calls

Reply:
Users will still have to implement native Optimizely track and activate method calls - the NotificationListener class is called only when either track or activate in Optimizely's native API is invoked. That said, I agree that we should only listen for "activate" events! Users will probably already have Segment track events set up to track user events and goal completions.